features/237-cms-validation/api.md

```diff
@@ -211,3 +214,105 @@ validations:
     min: 2000-01-01
     max: 2001-01-01
 ```
+
+### Null value handling
+
+Cases where `null` value appears:
+
+- value cleared for `Datetime`, `Reference`, `EmbeddedReference`, `Asset` field
+- value was originally `null` and not updated when save
```

Comment:
Sorry does it meant:
1. if the value was not originally `null` and not updated, the value will be the old value
2. if the value was originally `null` or not updated when save, the value will be `null`

Reply:
> if the value was not originally `null` and not updated, the value will be the old value

Yes.

To be more accurate, if the field is not updated, the value will be the old value, no matter what value it was.

> if the value was originally `null` or not updated when save, the value will be `null`

Do you mean:
if the value was originally `null` ~~or~~ and not updated when save, the value will be `null`


In cms, the empty value of each field is different, for example, empty value of string fields is empty string, while that of asset field is null. Thus some field will never produce a null value, once the cms user has edited the field.

At this moment, there is not a generic config for cms developer to specify the nullability a field, except dropdown.